src/Plugin.php

```diff
@@ -671,6 +677,35 @@ private function _registerElementExports(): void
         });
     }
 
+    /**
+     * Register Commerce related debug panels.
+     *
+     * @since 4.0
+     */
+    private function _registerDebugPanels(): void
+    {
+        Event::on(Application::class, Application::EVENT_BEFORE_REQUEST, function() {
+            /** @var Module|null $module */
+            $module = Craft::$app->getModule('debug');
+            $user = Craft::$app->getUser()->getIdentity();
+
+            if (!$module || !$user || !Craft::$app->getConfig()->getGeneral()->devMode) {
+                return;
+            }
+
+            $pref = Craft::$app->getRequest()->getIsCpRequest() ? 'enableDebugToolbarForCp' : 'enableDebugToolbarForSite';
+            if (!$user->getPreference($pref)) {
+                return;
+            }
+
+            $module->panels['commerce'] = new CommercePanel([
+                'id' => 'commerce',
+                'module' => $module,
+                'cart' => Plugin::getInstance()->getCarts()->getCart(),
```

Comment:
Since carts are just for front-ends should we only get/show the cart on front-end requests?

Reply:
Resolved with: https://github.com/craftcms/commerce/commit/e6af11b42fa624545ddceba07b2391837d3fb60f